api/internal/git/repospec.go

```diff
@@ -232,31 +232,63 @@ func parseHostSpec(n string) (string, string) {
 	// This prefix signaled go-getter to use the git protocol to fetch the url's contents.
 	// We still accept this prefix.
 	n, _ = trimPrefixIgnoreCase(n, "git::")
-	// We support gh: assuming authors use it as a github shorthand, specified in .gitconfig.
-	if rest, found := trimPrefixIgnoreCase(n, "gh:"); found {
-		return "gh:", rest
+
+	scheme, n := parseScheme(n)
+	if scheme == "file://" || scheme == "gh:" {
+		// We support gh: assuming authors use it as a github shorthand, specified in .gitconfig.
+		// The file protocol specifies an absolute path to a local git repo. There is no host.
+		// In both cases, we return the scheme and the rest of the url.
+		return scheme, n
+	}
+
+	username, n := parseUsername(n)
```

Comment:
I think it is not always immediately obvious how `n` is changing throughout this code, so some communication that parts of n are being removed as we parse through it may be helpful. Wydt about renaming this to `trimUsername` (and renaming the above `parseScheme` to `trimScheme`)?

Reply:
Looking through the existing occurrences of `Trim[Thing]` in this file, I think we have three operations we can distinguish:
* `TrimThing(input string)(rest string)` (or `TrimThing(input string)(rest string, trimmed bool)`) -- "trim" is used for this in stdlib functions too, like `strings.TrimPrefix`, which also removes a known quantity and returns the result (not the thing being trimmed)
* `ParseThing(input string)(thing string)` -- returns the target without reference to the remainder of the input
* `??Thing(input string)(thing string, rest string)` <-- this case. Since "trim" usually removes the thing in question without returning it, I'm leaning towards a third verb here. I'm thinking "extract" but I'm not set on it.

Reply:
Extract sgtm!